### PR TITLE
client.ImagePush(): default to ":latest" instead of "all tags"

### PR DESCRIPTION
**Pull request by thaJeztah**
relates to https://github.com/docker/cli/issues/2214 and https://github.com/docker/cli/pull/2220

The `docker push` command up until docker v0.9.1 always pushed all tags of a given
image, so `docker push foo/bar` would push (e.g.) all of  `foo/bar:latest`, `foo:/bar:v1`,
and `foo/bar:v1.0.0`.

Pushing all tags of an image was not desirable in many case, so docker v0.10.0
enhanced `docker push` to optionally specify a tag to push (`docker push foo/bar:v1`)
(see issue https://github.com/moby/moby/issues/3411 and https://github.com/moby/moby/pull/4948 (commit e648a186d68dcb3ee0d6123b041c5aa66438cc89).

This behavior exists up until today, and is confusing, because unlike other commands,
`docker push` does not default to use the `:latest` tag when omitted, but instead
makes it push "all tags of the image".

`docker pull` had a similar behavior, but https://github.com/moby/moby/pull/7759 (9c08364a412a51c80e8d17ae14f92549dc543e68)
changed the behavior to default to the `:latest` tag, and added a `--all-tags` flag
to the CLI to optionally pull all images.

This patch implements the API client changes to make `docker push` match the behavior
of `docker pull`, and default to pull a single image, unless the `all` option is passed.

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: client/image_push.go [hunk 1]
```diff
@@ -25,15 +25,14 @@ func (cli *Client) ImagePush(ctx context.Context, image string, options types.Im
...
-	tag := ""
 	name := reference.FamiliarName(ref)
-
-	if nameTaggedRef, isNamedTagged := ref.(reference.NamedTagged); isNamedTagged {
-		tag = nameTaggedRef.Tag()
-	}
-
 	query := url.Values{}
-	query.Set("tag", tag)
+	if !options.All {
```
Comment (thaJeztah) [inline client/image_push.go]: I was considering if we should produce an error if `options.All` is set and `image` contains a tag; currently this situation is handled in the CLI;

```go
	case opts.all && !reference.IsNameOnly(ref):
 		return errors.New("tag can't be used with --all-tags/-a")
```

Which is the same as how it's handled for `docker pull`, but perhaps we want that case to be handled by the client instead.

Also happy to do that in a follow-up and do it for both `Push` and `Pull` if we want them to act consistent.

**Comment by thaJeztah** [comment]
ping @tianon @dmcgowan @tiborvass PTAL

**Review by tianon — APPROVED** [review]
IMO this is fine as-is (given the edge case is not likely common); LGTM

**Review by tonistiigi — APPROVED** [review]

**Review by dmcgowan — APPROVED** [review]
LGTM

**Pull request merged**